### PR TITLE
Held Items

### LINKED ISSUE #281: Items still missing consumable flag
Not sure if this is the only case, but Focus Sash is missing a `FlagConsumable` which is causing errors in my testing of #211 

Related: #204

### PR DESCRIPTION
**Pull request by adapap**
See [held items section](https://veekun.com/dex/items/misc).

All items below implemented, checklist is for tests implemented:
- [x] Black Sludge
- [x] Damp Rock
- [x] Destiny Knot
- [x] Focus Sash
- [x] Heat Rock
- [x] Icy Rock
- [x] King's Rock
- [x] Leftovers
- [x] Life Orb
- [x] Mental Herb
- [x] Muscle Band
- [x] Razor Claw
- [x] Scope Lens
- [x] Shell Bell
- [x] Smooth Rock
- [x] White Herb
- [ ] Wide Lens
- [x] Wise Glasses

## Moved
#243 #244 #245 #246 #247 #248 #249 #250 #251 #252 #253 #255 

This PR also helps me figure out what features need to still be implemented in the "core" of the library.
## Blockers
~#85 (Effectiveness in damage formula) - Expert Belt~

closes #281 (changed for this PR)

### CODE REVIEW

**Comment by dyc3** [comment]
It might be a good idea to split this up into multiple PRs, maybe multiple issues. At least for the ones that are blocked by other issues.

**Comment by adapap** [comment]
Added issues for every item that cannot be immediately implemented. No tests are included yet, just want a review on functionality.

File: battle.go
Comment (dyc3) [inline battle.go:159]: This won't work correctly because the RNG is being rolled multiple times.

**Review by dyc3 — CHANGES_REQUESTED** [review]
Just one thing needs to be fixed

File: battle.go
Comment (adapap) [reply battle.go:159]: Going to move this into a separate issue

File: move.go
Comment (dyc3) [inline move.go:75]: These should probably just go in the `Move` struct (or `MoveData` in PR #256 )

File: transactions.go
Comment (dyc3) [inline transactions.go:250]: Leave this as `target`

**Review by dyc3 — CHANGES_REQUESTED** [review]
Oof owie my test coverage. More tests need to be added.

**Comment by dyc3** [comment]
I added a couple of tests. Looks like the Focus Sash doesn't have the consumable flag set, as a result of #204. We should probably fix that before merging this in.

File: transactions.go
Comment (adapap) [reply transactions.go]: A lot of cases exist where something happens as a fraction of Max HP, with a minimum of 1. This is probably a separate issue that extends beyond just this item.

**Comment by adapap** [comment]
Added all the tests that I could reasonably add. Some blockers/challenges include:

- Some items, as mentioned in #281, are not "consumable" so these tests are failing
- We don't have a way to test for effective accuracy in a battle so there is no way to test "Wide Lens" at the moment -> it would probably need accuracy calculation to be moved outside of damage calculation
- Had a few bugs with transaction utils with Gomega so I attempted to fix them. Also wanted to make the diffing look nicer so made a few QoL changes.

File: calc.go
Comment (dyc3) [inline calc.go:3]: Don't change this function signature, it intentionally doesn't take a `*Battle` and returns a `uint`

File: calc.go
Comment (dyc3) [inline calc.go:12]: This was intentionally moved outside of this function because it's random. Please put it back where it was.

File: data/gen.go
Comment (dyc3) [inline data/gen.go:58]: This is just more move data, it should just go in `data_move`.

File: data/gen.go
Comment (dyc3) [inline data/gen.go:604]: There is no need to have a separate struct `MoveMeta` inside of `MoveData`. Just make all the fields of `MoveMeta` fields of `MoveData` instead.

Also, the format string `%#v` outputs the struct as go code.

File: calc.go
Comment (dyc3) [inline calc.go:62]: Would prefer these to be integer operations

**Review by dyc3 — CHANGES_REQUESTED** [review]

File: util_test.go
Comment (dyc3) [inline util_test.go:335]: Could you move this change to a separate PR?

File: calc.go
Comment (adapap) [reply calc.go:62]: How do you do integer operations? Best way i've come up with is:
```golang
x *= 1.35
// in integer operation
x = (x * 135) / 100
```

Can we add some utility functions to perform floating/percentage modifiers on `uint` values?

Comment (dyc3) [reply calc.go:62]: Yeah, that's basically it. I don't think it would be very beneficial to add utility functions just to do a multiply and divide

File: battle.go
Comment (dyc3) [inline battle.go:136]: Change this back, it breaks sleeping for pokemon that are not in party 0 slot 0

File: battle_test.go
Comment (dyc3) [inline battle_test.go:1052]: Needs to assert that there was no `FaintTransaction`

File: move.go
Comment (dyc3) [reply move.go:81]: These should go in `MoveData` directly, and the `MoveMeta` type should not be necessary.

**Review by dyc3 — CHANGES_REQUESTED** [review]
Let's try to avoid big PRs like this in the future.

File: transactions.go
Comment (dyc3) [inline transactions.go:224]:
```suggestion
	t.Move.CurrentPP += uint8(t.Amount)
	if t.Move.CurrentPP >= t.Move.MaxPP {
		if t.Amount > 0 {
			t.Move.CurrentPP = t.Move.MaxPP
		} else {
			t.Move.CurrentPP = 0
		}
	}
```

**Review by dyc3 — CHANGES_REQUESTED** [review]

**Review by dyc3 — APPROVED** [review]
I think we're good to go on this. All we need to do is fix the merge conflicts.

Definitely make these smaller in the future though, this took way too long to merge in

**Pull request merged**
